tests/aws/services/events/helper_functions.py

```diff
@@ -7,14 +7,14 @@
 
 
 def is_v2_provider():
-    return os.environ.get("PROVIDER_OVERRIDE_EVENTS") == "v2" and not is_aws_cloud()
+    return (
+        os.environ.get("PROVIDER_OVERRIDE_EVENTS", "") not in ("v1", "legacy")
+        and not is_aws_cloud()
```

Comment:
```suggestion
        and not is_aws_cloud()
```
nit: not a fan of that combination here